include/envoy/network/address.h

```diff
@@ -85,6 +85,11 @@ class Ip {
    * @return the version of IP address.
    */
   virtual IpVersion version() const PURE;
+
+  /**
+   * @return IPv4-IPv6 mapping disabled for IPv6 addresses?
+   */
+  virtual bool v6only() const PURE;
```

Comment:
Should this be on class Ipv6 instead of here?

Reply:
Maybe? Need to do some dynamic casting to make use of where I needed it, but I agree it's cleaner at the interface level.

Reply:
Can't you pivot off of ipv6() being not-null in the cases that you need?